server/manager/src/main/java/org/apache/accumulo/manager/tableOps/delete/DeleteTable.java

```diff
@@ -41,6 +42,19 @@ public DeleteTable(NamespaceId namespaceId, TableId tableId) {
 
   @Override
   public long isReady(long tid, Manager env) throws Exception {
+
+    // Before attempting to delete the table, cancel any running user
+    // compactions.
+    if (Utils.reserveNamespace(env, namespaceId, tid, false, true, TableOperation.COMPACT_CANCEL)
+        + Utils.reserveTable(env, tableId, tid, false, true, TableOperation.COMPACT_CANCEL) == 0) {
+      try {
+        CancelCompactions.mutateZooKeeper(tid, tableId, env);
+      } finally {
+        Utils.unreserveTable(env, tableId, tid, false);
+        Utils.unreserveNamespace(env, namespaceId, tid, false);
+      }
+    }
+
     return Utils.reserveNamespace(env, namespaceId, tid, false, false, TableOperation.DELETE)
         + Utils.reserveTable(env, tableId, tid, true, true, TableOperation.DELETE);
```

Comment:
Not related to your changes, but is it weird that reserveNamespace is okay with the namespace not existing, but reserveTable requires the table to exist? How can the table exist without its namespace? And, would we want to try to delete a table that's in a non-existent namespace?